extensions/eo/examples/landsat8-collection.json

```diff
@@ -0,0 +1,229 @@
+                "producer"
+            ],
+            "url": "https://landsat.usgs.gov/"
+        },
+        {
+            "name": "AWS",
+            "roles": [
+                "host"
+            ],
+            "url": "https://landsatonaws.com/"
+        }
+    ],
+    "license": "PDDL-1.0",
+    "summaries": {
+        "eo:gsd": [15],
+        "platform": ["landsat-4", "landsat-5", "landsat-7", "landsat-8"],
+        "eo:instruments": ["OLI, TIRS"],
+        "view:off_nadir": [0]
+    },
+    "eo:bands": [
+        {
+            "name": "L8-B1",
+            "common_name": "coastal",
+            "gsd": 30,
+            "center_wavelength": 0.44,
+            "full_width_half_max": 0.02
+        },
+        {
+            "name": "L8-B2",
+            "common_name": "blue",
+            "gsd": 30,
+            "center_wavelength": 0.48,
+            "full_width_half_max": 0.06
+        },
+        {
+            "name": "L8-B3",
+            "common_name": "green",
+            "gsd": 30,
+            "center_wavelength": 0.56,
+            "full_width_half_max": 0.06
+        },
+        {
+            "name": "L8-B4",
+            "common_name": "red",
+            "gsd": 30,
+            "center_wavelength": 0.65,
+            "full_width_half_max": 0.04
+        },
+        {
+            "name": "L8-B5",
+            "common_name": "nir",
+            "gsd": 30,
+            "center_wavelength": 0.86,
+            "full_width_half_max": 0.03
+        },
+        {
+            "name": "L8-B6",
+            "common_name": "swir16",
+            "gsd": 30,
+            "center_wavelength": 1.6,
+            "full_width_half_max": 0.08
+        },
+        {
+            "name": "L8-B7",
+            "common_name": "swir22",
+            "gsd": 30,
+            "center_wavelength": 2.2,
+            "full_width_half_max": 0.2
+        },
+        {
+            "name": "L8-B8",
+            "common_name": "pan",
+            "gsd": 15,
+            "center_wavelength": 0.59,
+            "full_width_half_max": 0.18
+        },
+        {
+            "name": "L8-B9",
+            "common_name": "cirrus",
+            "gsd": 30,
+            "center_wavelength": 1.37,
+            "full_width_half_max": 0.02
+        },
+        {
+            "name": "L8-B10",
+            "common_name": "lwir11",
+            "gsd": 100,
+            "center_wavelength": 10.9,
+            "full_width_half_max": 0.8
+        },
+        {
+            "name": "L8-B11",
+            "common_name": "lwir12",
+            "gsd": 100,
+            "center_wavelength": 12,
+            "full_width_half_max": 1
+        }
+    ],
+    "assets": {
+        "index": {
+            "type": "text/html",
+            "title": "HTML index page"
+        },
+        "thumbnail": {
+            "title": "Thumbnail image",
+            "type": "image/jpeg"
+        },
+        "B1": {
+            "type": "image/x.geotiff",
+            "eo:bands": [
```

Comment:
Now that I'm seeing this, I'm really wondering why we don't combine assets and eo extensions to make something that's more expressive. Like wouldn't it be charming to have the actual band objects here and then the Items could inherit that? Would avoid all those array indices.

Reply:
Yes! I'm all for that.   I think I had brought this up at some point. I'd like to see the Asset definition made core (not an extension), then the eo:bands appears under each asset and is an array of Band objects directly rather than indices.

Would be so much more manageable and readable. There might be some redundancy if there are multiple assets that share a band, but it's not much.

eo:bands would only be used in the asset definition, so in order to find out band ordering for an asset in an item, you would need to get the collection.  I'm fine with that, I think it's better than having indices in the item that need to be looked up in the collection.

Reply:
I'd like that. I also feel assets (with a new name, see #779) should be core. As with summaries, the utility is high and just makes Collections really useful.

Reply:
No, that's the nice thing about calling this an "asset definition" rather than assets, it can have different fields than assets in items.

We could also move other fields to the asset definition as well...we can discuss, we wouldn't want assets to just be reduced to an href.

I wouldn't want to have to go look up `type` in the asset definition for instance, but something like `roles` might be better off there.

Reply:
It was not clear to me that you are allowed to add more than actually available in the item assets. I thought it's a strict copy of the available assets excluding the href.

What fields are you thinking about to move?

Reply:
Well you can't do that right now - the asset extension is simply just a copy of all the same fields in the assets, but we could define the asset definition differently.

The point of the asset definition is to indicate what assets are going to be available in a member item, so I could actually see all of the fields getting included in the asset definition. It would be good to know the title, description, type, and roles for each of the assets.

But at the same time I wouldn't want all that stuff removed from the Item assets...nor would I want to have to repeat all that same information so I'm not sure there's a straightforward answer here.